### PR TITLE
Meshservice feature

### PR DESCRIPTION
**Pull request by lrodri29**
Adds local meshing service using zmesh. 
```python
from intern.remote.boss import BossRemote
from intern.service.mesh.service import MeshService, VoxelUnits
import numpy as np

rmt = BossRemote({"protocol": "https", "host": "api.bossdb.io", "token": "public",})

COLL = "kharris15"
EXP = "apical"
CHAN = "anno"
MESH_ID = 22

res = 0
x_rng = [4500, 4700]
y_rng = [3600, 3900]
z_rng = [0, 162]

###### Use MeshService from remote ######

# Define channel resource
ann_chan = rmt.get_channel(CHAN, COLL, EXP)

# Use resource to generate mesh from the Segmentation ID specified
# and the specified cutout volume ranges
# Save the output in a tuple, raw_data and mesh_data will be returned.
mesh = rmt.mesh(
    ann_chan,
    res,
    x_rng,
    y_rng,
    z_rng,
    id_list=[MESH_ID],
    voxel_unit=VoxelUnits.um,
    simp_fact=100,
)

# Convert mesh data to obj
mesh_obj = mesh.obj_mesh()

# Convert mesh data to precompute format for neuroglancer
mesh_ng = mesh.ng_mesh()

# Write the mesh obj
with open("mesh_22_test.obj", "wb") as fh:
    fh.write(mesh_obj)

# Write the Neuroglancer ready mesh
with open("mesh_ng_test", "wb") as fh:
    fh.write(mesh_ng)
```

### CODE REVIEW

File: intern/remote/remote.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import six
 from abc import ABCMeta
 from six.moves import configparser
+from intern.service.mesh.service import MeshService
```
Comment (j6k4m8) [inline intern/remote/remote.py:17]: we should convenience-import this so it's only `from intern.service import MeshService`. (I can show how to do this if helpful)

File: intern/service/mesh/service.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
```
Comment (j6k4m8) [inline intern/service/mesh/service.py:1]: these can all be 2020 in the new/edited files

File: examples/meshing_example.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+volume = rmt.get_cutout(ann_chan, res, x_rng, y_rng, z_rng)
+
+# Initialize MeshService
+mesh_serv = MeshService()
+
+# Create mesh
+mesh = mesh_serv.create(volume, x_rng, y_rng, z_rng)
+
+# Convert mesh data to precompute format for neuroglancer
+mesh_obj = rmt.obj_mesh(mesh)
```
[the remark this answers is not in the document]
Comment (j6k4m8) [reply examples/meshing_example.py:61]: to elaborate on this:

`MeshService.create` returns a `Mesh` object, with something like the following interface:

```
class Mesh:
    def __init__(self, mesh: Tuple[List, List]): # `mesh` as returned from the current `MeshService.create`
        ...

    def ng_mesh(self) -> precompute_format:
        ...

    ...etc
```

so you can run the following code:

```python
new_mesh = my_remote.mesh(...)
# OR:
new_mesh = mesh_service.create(...)

mesh_obj = new_mesh.to_obj()
```

File: examples/meshing_example.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+from intern.remote.boss import BossRemote
+from intern.service.mesh.service import MeshService
+import numpy as np
+
+rmt = BossRemote({
+"protocol": "https",
+"host": "api.bossdb.io",
+"token": "<token>",
```
[the remark this answers is not in the document]
Comment (lrodri29) [reply examples/meshing_example.py:8]: Very true

File: examples/meshing_example.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
+from intern.remote.boss import BossRemote
+from intern.service.mesh.service import MeshService
+import numpy as np
+
+rmt = BossRemote({
+"protocol": "https",
```
[the remark this answers is not in the document]
Comment (lrodri29) [reply examples/meshing_example.py:6]: is this magic?

File: examples/meshing_example.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ann_chan = rmt.get_channel(CHAN, COLL, EXP)
+
+# Use resource to generate mesh from the Segmentation ID specified
+# and the specified cutout volume ranges
+# Save the output in a touple, raw_data and mesh_data will be returned. 
+raw_data, mesh_data = rmt.mesh(ann_chan, res, x_rng, y_rng, z_rng, 
+    id_list=[MESH_ID], voxel_unit="micrometers", simp_fact=100)
+
+# Convert mesh data to obj
+mesh_obj = rmt.obj_mesh(mesh_data)
```
[the remark this answers is not in the document]
Comment (lrodri29) [reply examples/meshing_example.py:33]: Yeh yeh! I like this

File: intern/service/mesh/service.py [hunk 7]
```diff
@@ -160,4 +149,16 @@ def obj_mesh(self):
             (): Returns mesh obj format
 
         """
-        return self._mesh.to_obj()
\ No newline at end of file
+        return self._mesh.to_obj()
+
+class VoxelUnits(IntEnum):
+    """Enum with valid VoxelUnits
+    """
```
Comment (j6k4m8) [inline intern/service/mesh/service.py:156]: this set of lines where every line is one longer than its predecessor is the most soothing thing i've ever seen

Comment (lrodri29) [reply intern/service/mesh/service.py]: 👯

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ---
+## v?.?.?
+> Addition of an intern MeshService.
```
Comment (j6k4m8) [inline CHANGELOG.md:5]: You can merge this with the next version header below as another Feature bullet :)

Comment (j6k4m8) [reply CHANGELOG.md:5]: Good to merge once that's fixed!

Comment (lrodri29) [reply CHANGELOG.md]: done

**Pull request merged**